### PR TITLE
chore: init models for v3 spec

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Init models and types for `v3` spec to enable contributions.

cc @smoya

### CODE REVIEW

File: src/models/v3/bindings.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import type  { v2 } from '../../spec-types';
+
+export class Bindings extends Collection<BindingInterface> implements BindingsInterface {
+  override get<T extends Record<string, any> = Record<string, any>>(name: string): BindingInterface<T> | undefined {
+    return this.collections.find(binding => binding.protocol() === name);
+  }
+
+  extensions(): ExtensionsInterface {
+    const extensions: ExtensionInterface[] = [];
+    Object.entries(this._meta.originalData as v2.SpecificationExtensions || {}).forEach(([id, value]) => {
```
Comment (jonaslagoni) [inline src/models/v3/bindings.ts:22]: Should this not be `v3`? :thinking:

File: src/models/v3/extension.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { BaseModel } from '../base';
+
+import type { ExtensionInterface } from '../extension';
+
+import type { v2 } from '../../spec-types';
```
Comment (jonaslagoni) [inline src/models/v3/extension.ts:5]: v3?

File: src/models/v3/external-docs.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { BaseModel } from '../base';
+
+import { hasDescription, description, extensions } from './mixins';
+
+import type { ExternalDocumentationInterface } from '../external-docs';
+import type { ExtensionsInterface } from '../extensions';
+
+import type { v2 } from '../../spec-types';
```
Comment (jonaslagoni) [inline src/models/v3/external-docs.ts:8]: v3?

File: src/models/v3/tag.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { BaseModel } from '../base';
+
+import { hasDescription, description, extensions, hasExternalDocs, externalDocs } from './mixins';
+
+import type { ExtensionsInterface } from '../extensions';
+import  type{ ExternalDocumentationInterface } from '../external-docs';
+import type { TagInterface } from '../tag';
+
+import type { v2 } from '../../spec-types';
```
Comment (jonaslagoni) [inline src/models/v3/tag.ts:9]: v3?

**Review by jonaslagoni — CHANGES_REQUESTED** [review]

File: src/models/v3/bindings.ts (hunk 1 above)
Comment (magicmatatjahu) [reply src/models/v3/bindings.ts]: Yeah, it should be `v3`, thanks! :)

Comment (magicmatatjahu) [reply src/models/v3/bindings.ts]: Fixed!

**Review by jonaslagoni — APPROVED** [review]

**Pull request merged**